java-manta-client-unshaded/src/main/java/com/joyent/manta/http/MantaHttpRequestFactory.java

```diff
@@ -229,6 +229,10 @@ private void prepare(final HttpRequest request) {
     public static void addHeaders(final HttpMessage httpMessage, final Header... headers) {
         Validate.notNull(httpMessage, "HttpMessage must not be null");
         for (final Header header : headers) {
+            if (header.getName().equals(MantaHttpHeaders.REQUEST_ID)) {
```

Comment:
As a paranoia sufferer, I would invert the equals to avoid the remote possibility of a NPE:
```java
if (MantaHttpHeaders.REQUEST_ID.equals(header.getName())) {
```

Reply:
Thanks, I was only taking existing calls to `addHeaders` into account.